### DISCUSSION TITLE
Issue with style.json and tiles vector URL

### DISCUSSION THREAD

**Discussion by raphael10-collab** [maplibre/maplibre-gl-js#4417] [Q&A]
As explained and asked in Slack Forum : https://osmus.slack.com/archives/C01G3D28DAB/p1721219062146369?thread_ts=1721213195.422509&cid=C01G3D28DAB 

I want to use this TileServer: https://tileserver.linuxbabe.com/ with MapLibre

The style.json should be this: https://tileserver.linuxbabe.com/styles/basic-preview/style.json
and, based on https://tileserver.linuxbabe.com/data/v3.json  
![image](https://github.com/user-attachments/assets/4a4727a3-e118-4ddc-a58f-b2ac10df6c7b)  ,

the tiles vector URL should be this: https://tileserver.linuxbabe.com/data/v3/{z}/{x}/{y}.pbf

But If I put those two URLs in their fields in the HTML, I do not get any output : 

![image](https://github.com/user-attachments/assets/5a37c3b6-bdaf-4a65-9bb1-471bfec296e2)

This is the complete `map.html` file :  

    <!DOCTYPE html>
    <html>
        <head>
             <meta charset="utf-8">
             <title>Vector Tile Map with MapLibre</title>
             <meta name="viewport" content="initial-scale=1,maximum-scale=1,user-scalable=no">
             <script src='https://unpkg.com/maplibre-gl@latest/dist/maplibre-gl.js'></script>
            <link href='https://unpkg.com/maplibre-gl@latest/dist/maplibre-gl.css' rel='stylesheet' />
             <style>
                body { margin: 0; padding: 0; }
                #map { position: absolute; top: 0; bottom: 0; width: 100%; }
             </style>
       </head>
       <body>
            <div id="map"></div>
            <script>
                var map = new maplibregl.Map({
                container: 'map',
                style:
                        'https://tileserver.linuxbabe.com/styles/basic-preview/style.json',
                zoom: 5.43,
                center: [-3.9, 54.5]
        });

        map.on('load', function () {
                  // Add a new vector tile source with ID 'linuxbabe'.
                  map.addSource('linuxbabe', {
                         'type': 'vector',
                         'tiles': [
                              'https://tileserver.linuxbabe.com/data/v3/{z}/{x}/{y}.pbf'
                          ],
                          'minzoom': 6,
                          'maxzoom': 14
                    });
                    map.addLayer(
                       {
                            'id': 'default', // Layer ID
                            'type': 'line',
                            'source': 'linuxbabe', // ID of the tile source created above
                               // Source has several layers. We visualize the one with name 'sequence'.
                            'source-layer': 'sequence',
                            'layout': {
                                      'line-cap': 'round',
                                      'line-join': 'round'
                             },
                            'paint': {
                                       'line-opacity': 0.6,
                                       'line-color': 'rgb(53, 175, 109)',
                                       'line-width': 2
                             }
                      },
                   );
                });

                map.addControl(new maplibregl.NavigationControl());

                //set max zoom level (0-24)
                map.setMaxZoom(19);
        </script>
        </body>
</html>

What am I missing and / or am I doing wrong?

**Comment by HarelM** [comment]
Can you please share a jsbin/codepen with the relevant code?

**Comment by raphael10-collab** [comment]
Hi @HarelM ! Thank you for helping

jsfiddle: https://jsfiddle.net/RaphyCollab/oq19abd4/2/
And here a jsbin repo : https://jsbin.com/dekujuz/edit?html,output
github gist : https://gist.github.com/raphael10-collab/2c1676a37c2193c79b110ecdfe33215d

I've taken the HTML and Javascript code from `"Display Map with MapLibre GL"` in  https://www.linuxbabe.com/ubuntu/basemap-tileserver-gl-openmaptiles-ubuntu-22-04 , and I took the https://tileserver.linuxbabe.com/styles/basic-preview/style.json URL and https://tileserver.linuxbabe.com/data/v3/{z}/{x}/{y}.pbf URL from https://tileserver.linuxbabe.com/

@HarelM What is missing and/or wrong?
